src/Microsoft.Management.Deployment/PackageCatalogReference.cpp

```diff
@@ -54,6 +55,11 @@ namespace winrt::Microsoft::Management::Deployment::implementation
                 return GetConnectCatalogErrorResult();
             }
 
+            if (!m_acceptSourceAgreements)
+            {
+                return GetConnectCatalogErrorResult();

```

Comment:
Should we fail only if !AcceptAgreements && Agreements exist?

Reply:
Changed condition to suggested